### DISCUSSION TITLE
A simple and self-constrained example that using the augurs npm package

### DISCUSSION THREAD

**Discussion by shenxiangzhuang** [grafana/augurs#190] [Q&A] [answered]
Hi, thanks for your awesome work firstly! I'm wondering is there a simple and self-constrained example that using the npm package [augurs](https://www.npmjs.com/package/@bsull/augurs?activeTab=readme)? 

I want to know how to use this package in a new javascript project. I see the example at https://github.com/grafana/augurs/blob/main/demo/clustering.js, but which is using the file in dist rather than the npm package. Sorry for that I'm not familiar with javascript, so I don't know how to transform the example from "dist files" to  "npm package". Any  suggestion will be appreciated!

**Comment by sd2k** [comment, marked as answer]
This is definitely an area that's lacking! I'll make a note to update the README for the npm package.

In the meantime there is a [test package](https://github.com/grafana/augurs/tree/main/js%2Ftestpkg) which imports from the npm package which might be helpful. Let me know if that's any use.

**Reply by shenxiangzhuang** [reply]
Thanks for your timely reply, and the test package helps a lot! I can use the package and run it now.

There is another minor question about the code:
https://github.com/grafana/augurs/blob/9652eba06fa4494017c7d93b4d90323b53e27df4/js/testpkg/clustering.test.ts#L1-L9

It seems we have to specify the wasm file path by hand when we use it, for example ` initClusteringSync({ module: readFileSync('node_modules/@bsull/augurs/clustering_bg.wasm') }); `. I'm wondering is there any other more convenient method that we can use to load the wasm module?

**Reply by sd2k** [reply]
Ah, ordinarily you shouldn't have to do that, that's just there because I wanted to set up the tests more explicitly.

In most cases you can do something like:

```ts
import init, { Dtw } from '@bsull/augurs/dtw';

async function run() {
    await init().then(() => console.log('augurs dtw initialized'));

    Dtw.euclidean().distanceMatrix([ ... ]);
}
```

The `init` async function should handle loading and initializing the WASM for you using `import.meta` (at least in a browser context).

**Reply by shenxiangzhuang** [reply]
I got it. Thanks a lot!
